### ISSUE TITLE
Unable to delegate from windows to linux

### ISSUE THREAD

**Issue by t-v** [ansible/ansible#14677]
##### Issue Type:
- Bug Report
##### Ansible Version:

```
ansible 2.0.1.0
  config file = 
  configured module search path = Default w/o overrides
```
##### Ansible Configuration:

No config file, only these things are put in my environment variables

```
ANSIBLE_SSH_ARGS="-o ControlMaster=no -o StrictHostKeyChecking=no -o BatchMode=yes"
ANSIBLE_TRANSPORT="ssh"
ANSIBLE_VAULT_PASSWORD_FILE="~/git/operations/werner/vault/decrypt-vault"
```
##### Environment:

CentOS6
##### Summary:

When delegating an action for a windows host to a linux host I get this error:

```
FAILED! => {"failed": true, "msg": "ssl: 500 WinRMTransport. [Errno 111] Connection refused"}
```
##### Steps To Reproduce:

Try delegating an action from a windows host to a linux host

my hosts file:

```
acc-varnish01-atr
acc-varnish02-blo

[win2012]
acc-rt1903-sta
acc-rt1903-web1
acc-rt1903-web2
```

and in accept/group_vars/win2012.yml

``` yaml
ansible_ssh_user: Ansible
ansible_ssh_pass: '*****'
ansible_ssh_port: 5986
ansible_connection: winrm
```

my playbook:

``` yaml
- hosts: acc-rt1903-sta:acc-varnish01-atr
  tasks:
    - name: test win_ping to windows host
      action: win_ping
      when: inventory_hostname in groups['win2012']

    - name: test ping to linux host
      action: ping
      when: inventory_hostname not in groups['win2012']

    - debug: var=ansible_connection    
      delegate_to: acc-rt1903-web1

    - debug: var=ansible_connection
      delegate_to: acc-varnish02-blo

    - name: win_ping when windows
      action: win_ping
      delegate_to: acc-rt1903-web1
      when: inventory_hostname in groups['win2012']

    - name: win_ping when linux
      action: win_ping
      delegate_to: acc-rt1903-web1
      when: inventory_hostname not in groups['win2012']

    - name: ping when linux
      action: ping
      delegate_to: acc-varnish02-blo
      when: inventory_hostname not in groups['win2012']

    - name: ping when windows
      action: ping
      delegate_to: acc-varnish02-blo
      when: inventory_hostname in groups['win2012']
```
##### Expected Results:

When I used to use ansible 1.9.2 with fix https://github.com/ansible/ansible/issues/13561, I get from this exact playbook

```
> ansible-playbook -i accept/ test.yml

PLAY [acc-rt1903-sta:acc-varnish01-atr] *************************************** 

GATHERING FACTS *************************************************************** 
ok: [acc-varnish01-atr]
ok: [acc-rt1903-sta]

TASK: [test win_ping to windows host] ***************************************** 
skipping: [acc-varnish01-atr]
ok: [acc-rt1903-sta]

TASK: [test ping to linux host] *********************************************** 
skipping: [acc-rt1903-sta]
ok: [acc-varnish01-atr]

TASK: [debug var=ansible_connection] ****************************************** 
ok: [acc-rt1903-sta -> acc-rt1903-web1] => {
    "var": {
        "ansible_connection": "winrm"
    }
}
ok: [acc-varnish01-atr -> acc-rt1903-web1] => {
    "var": {
        "ansible_connection": "ansible_connection"
    }
}

TASK: [debug var=ansible_connection] ****************************************** 
ok: [acc-rt1903-sta -> acc-varnish02-blo] => {
    "var": {
        "ansible_connection": "winrm"
    }
}
ok: [acc-varnish01-atr -> acc-varnish02-blo] => {
    "var": {
        "ansible_connection": "ansible_connection"
    }
}

TASK: [win_ping when windows] ************************************************* 
skipping: [acc-varnish01-atr]
ok: [acc-rt1903-sta -> acc-rt1903-web1]

TASK: [win_ping when linux] *************************************************** 
skipping: [acc-rt1903-sta]
ok: [acc-varnish01-atr -> acc-rt1903-web1]

TASK: [ping when linux] ******************************************************* 
skipping: [acc-rt1903-sta]
ok: [acc-varnish01-atr -> acc-varnish02-blo]

TASK: [ping when windows] ***************************************************** 
skipping: [acc-varnish01-atr]
ok: [acc-rt1903-sta -> acc-varnish02-blo]

PLAY RECAP ******************************************************************** 
acc-rt1903-sta             : ok=6    changed=0    unreachable=0    failed=0   
acc-varnish01-atr          : ok=6    changed=0    unreachable=0    failed=0 
```
##### Actual Results:

when I run this playbook on 2.x. 

```
> ansible-playbook -i accept/ test.yml

PLAY ***************************************************************************

TASK [setup] *******************************************************************
ok: [acc-varnish01-atr]
ok: [acc-rt1903-sta]

TASK [test win_ping to windows host] *******************************************
skipping: [acc-varnish01-atr]
ok: [acc-rt1903-sta]

TASK [test ping to linux host] *************************************************
skipping: [acc-rt1903-sta]
ok: [acc-varnish01-atr]

TASK [debug] *******************************************************************
ok: [acc-rt1903-sta -> acc-rt1903-web1] => {
    "ansible_connection": "winrm"
}
ok: [acc-varnish01-atr -> acc-rt1903-web1] => {
    "ansible_connection": "winrm"
}

TASK [debug] *******************************************************************
ok: [acc-varnish01-atr -> acc-varnish02-blo] => {
    "ansible_connection": "ssh"
}
ok: [acc-rt1903-sta -> acc-varnish02-blo] => {
    "ansible_connection": "winrm"
}

TASK [win_ping when windows] ***************************************************
skipping: [acc-varnish01-atr]
ok: [acc-rt1903-sta -> acc-rt1903-web1]

TASK [win_ping when linux] *****************************************************
skipping: [acc-rt1903-sta]
ok: [acc-varnish01-atr -> acc-rt1903-web1]

TASK [ping when linux] *********************************************************
skipping: [acc-rt1903-sta]
ok: [acc-varnish01-atr -> acc-varnish02-blo]

TASK [ping when windows] *******************************************************
skipping: [acc-varnish01-atr]
fatal: [acc-rt1903-sta]: FAILED! => {"failed": true, "msg": "ssl: 500 WinRMTransport. [Errno 111] Connection refused"}

NO MORE HOSTS LEFT *************************************************************

PLAY RECAP *********************************************************************
acc-rt1903-sta             : ok=5    changed=0    unreachable=0    failed=1   
acc-varnish01-atr          : ok=6    changed=0    unreachable=0    failed=0 
```

The [ping when windows] task in ultra verbose mode (but nothing really useful here):

```
TASK [ping when windows] *******************************************************
task path: /home/ansible/git/operations/werner-exh345/test.yml:34
skipping: [acc-varnish01-atr] => {"changed": false, "skip_reason": "Conditional check failed", "skipped": true}
<acc-varnish02-blo> ESTABLISH WINRM CONNECTION FOR USER: Ansible on PORT 5986 TO acc-varnish02-blo
fatal: [acc-rt1903-sta]: FAILED! => {"failed": true, "msg": "ssl: 500 WinRMTransport. [Errno 111] Connection refused"}
```

I created a new issue to put the issue in the new template and to focus some attention on it, since my last on has dropped quite low in the queue and this has begun to become quite blocking, so this was the former issue: https://github.com/ansible/ansible/issues/12370

**Comment by t-v** [comment]
Another thing I noticed is if I add
accpt/group_vars/all.yml:

``` yaml
ansible_connection: ssh
```

it changes my output. This means that the default isn't correctly implemented, because I only have the environment variable declaring the default as ssh and the windows group var that declares winrm for the machines in the win2012 group.

So this is the exact same run with the new file:

```
> ansible-playbook -i accept/ test.yml

PLAY ***************************************************************************

TASK [setup] *******************************************************************
ok: [acc-varnish01-atr]
ok: [acc-rt1903-sta]

TASK [test win_ping to windows host] *******************************************
skipping: [acc-varnish01-atr]
ok: [acc-rt1903-sta]

TASK [test ping to linux host] *************************************************
skipping: [acc-rt1903-sta]
ok: [acc-varnish01-atr]

TASK [debug] *******************************************************************
ok: [acc-varnish01-atr -> acc-rt1903-web1] => {
    "ansible_connection": "ssh"
}
ok: [acc-rt1903-sta -> acc-rt1903-web1] => {
    "ansible_connection": "winrm"
}

TASK [debug] *******************************************************************
ok: [acc-rt1903-sta -> acc-varnish02-blo] => {
    "ansible_connection": "winrm"
}
ok: [acc-varnish01-atr -> acc-varnish02-blo] => {
    "ansible_connection": "ssh"
}

TASK [win_ping when windows] ***************************************************
skipping: [acc-varnish01-atr]
ok: [acc-rt1903-sta -> acc-rt1903-web1]

TASK [win_ping when linux] *****************************************************
skipping: [acc-rt1903-sta]
ok: [acc-varnish01-atr -> acc-rt1903-web1]

TASK [ping when linux] *********************************************************
skipping: [acc-rt1903-sta]
ok: [acc-varnish01-atr -> acc-varnish02-blo]

TASK [ping when windows] *******************************************************
fatal: [acc-rt1903-sta]: FAILED! => {"failed": true, "msg": "to use the 'ssh' connection type with passwords, you must install the sshpass program"}
skipping: [acc-varnish01-atr]

NO MORE HOSTS LEFT *************************************************************

PLAY RECAP *********************************************************************
acc-rt1903-sta             : ok=5    changed=0    unreachable=0    failed=1   
acc-varnish01-atr          : ok=6    changed=0    unreachable=0    failed=0 
```

ultra verbose mode on the failing task:

```
TASK [ping when windows] *******************************************************
task path: /home/ansible/git/operations/werner-exh345/test.yml:34
<acc-varnish02-blo> ESTABLISH SSH CONNECTION FOR USER: Ansible
skipping: [acc-varnish01-atr] => {"changed": false, "skip_reason": "Conditional check failed", "skipped": true}
fatal: [acc-rt1903-sta]: FAILED! => {"failed": true, "msg": "to use the 'ssh' connection type with passwords, you must install the sshpass program"}
```

As you can see in the debug tasks, now it shows the correct connection protocol, but it seems it does not clean out the password, because only the windows hosts use a password, all the linux connections are authenticated by key.

**Comment by bcoca** [comment]
see if #15024  fixes this for your, it should prevent ansible_connection from being incorrectly assigned

**Comment by bcoca** [comment]
closing via merged #15024

**Comment by juanjomm** [comment]
### Problem Still reproducible after ansible upgrade:

**Ansible Version:**
`ansible 2.2.0 (devel c9a5b1c555) last updated 2016/06/02 15:42:56 (GMT +200)
  lib/ansible/modules/core: (detached HEAD ca4365b644) last updated 2016/06/02 15:43:14 (GMT +200)
  lib/ansible/modules/extras: (detached HEAD b0aec50b9a) last updated 2016/06/02 15:43:15 (GMT +200)
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides`

`Python 2.7.5 (default, Oct 11 2015, 17:47:16)
[GCC 4.8.3 20140911 (Red Hat 4.8.3-9)] on linux2`

`SO: RHEL 7.2`

**Ansible Configuration:**
`ansible_ssh_user: xxxx
 ansible_ssh_pass: xxxxx
 ansible_connection: winrm
 ansible_ssh_port: 5985`

**Summary:**
`When delegating an action for a windows host to a linux host I get this error:
 FAILED! => {"failed": true, "msg": "ssl: 500 WinRMTransport. [Errno 111] Connection refused"}`

**Steps To Reproduce:**
Try delegating an action from a windows host to a linux host:
file: main.yml
`- name: 01-EsperarTermineReinicioServidor
   wait_for: port=3389 host="192.168.1.10" delay=10 timeout=600 state=started
   delegate_to: 127.0.0.1`
